NUT-20: signature on mint request (#188)
cashubtc/nuts: error_codes.md

| 11002 | Transaction is not balanced (inputs != outputs)   | [NUT-02][02], [NUT-03][03], [NUT-05][05] |
| 11005 | Unit in request is not supported                  | [NUT-04][04], [NUT-05][05]               |
| 11006 | Amount outside of limit range                     | [NUT-04][04], [NUT-05][05]               |
| 12001 | Keyset is not known                               | [NUT-02][02], [NUT-04][04]               |
| 12002 | Keyset is inactive, cannot sign messages          | [NUT-02][02], [NUT-03][03], [NUT-04][04] |
| 20001 | Quote request is not paid                         | [NUT-04][04]                             |
| 20002 | Tokens have already been issued for quote         | [NUT-04][04]                             |
| 20003 | Minting is disabled                               | [NUT-04][04]                             |
| 20005 | Quote is pending                                  | [NUT-04][04], [NUT-05][05]               |
| 20006 | Invoice already paid                              | [NUT-05][05]                             |
| 20007 | Quote is expired                                  | [NUT-04][04], [NUT-05][05]               |

> | 20008 | Signature on mint request not provided or invalid | [NUT-20][20]                             |

Author: ```suggestion
| 20008 | Signature on mint request not provided or invalid | [NUT-20][20]                             |
| 20009 | Pubkey required on mint quote                              | [NUT-20][20]                             |
```

| 11002 | Transaction is not balanced (inputs != outputs)   | [NUT-02][02], [NUT-03][03], [NUT-05][05] |
| 11005 | Unit in request is not supported                  | [NUT-04][04], [NUT-05][05]               |
| 11006 | Amount outside of limit range                     | [NUT-04][04], [NUT-05][05]               |
| 12001 | Keyset is not known                               | [NUT-02][02], [NUT-04][04]               |
| 12002 | Keyset is inactive, cannot sign messages          | [NUT-02][02], [NUT-03][03], [NUT-04][04] |
| 20001 | Quote request is not paid                         | [NUT-04][04]                             |
| 20002 | Tokens have already been issued for quote         | [NUT-04][04]                             |
| 20003 | Minting is disabled                               | [NUT-04][04]                             |
| 20005 | Quote is pending                                  | [NUT-04][04], [NUT-05][05]               |
| 20006 | Invoice already paid                              | [NUT-05][05]                             |
| 20007 | Quote is expired                                  | [NUT-04][04], [NUT-05][05]               |

> | 20008 | Signature on mint request not provided or invalid | [NUT-19][19]                             |
> | 20009 | Pubkey required on mint quote                     | [NUT-19][19]                             |

Author: ```suggestion
| 20008 | Signature on mint request not provided or invalid | [NUT-20][20]                             |
```

| 20005 | Quote is pending                                | [NUT-04][04], [NUT-05][05]               |
| 20006 | Invoice already paid                            | [NUT-05][05]                             |
| 20007 | Quote is expired                                | [NUT-04][04], [NUT-05][05]               |

> | 20008 | Witness not provided for mint quote             | [NUT-19][19]                             |

Reviewer: We need an error for the invalid witness as well.

| 20005 | Quote is pending                                | [NUT-04][04], [NUT-05][05]               |
| 20006 | Invoice already paid                            | [NUT-05][05]                             |
| 20007 | Quote is expired                                | [NUT-04][04], [NUT-05][05]               |

> | 20008 | Invalid witness                                 | [NUT-19][19]                             |

Reviewer 2: If we want to make a `required` flag for the mint info, we would need an error code for the `Quote without pubkey` case

Reviewer 2: Maybe "invalid witness" and "no witness provided" can be the same error.

> | 20009 | Pubkey required for mint quote                  | [NUT-20][20]                             |

Reviewer: Is this error still relevant?

Author: Will be used for BOLT12 and onchain

[10]: 10.md
[11]: 11.md
[12]: 12.md

> [19]: 19.md

Author: ```suggestion
[20]: 20.md
```
